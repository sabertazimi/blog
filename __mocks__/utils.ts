let navigatorClipboardMocked = false
const writeTextMock = jest.fn()

const mockNavigatorClipboard = () => {
  if (!navigatorClipboardMocked) {
    Object.defineProperty(global.navigator, 'clipboard', {
      value: {
        writeText: writeTextMock,
      },
    })

    navigatorClipboardMocked = true
  }

  return { writeTextMock }
}

export { mockNavigatorClipboard }
