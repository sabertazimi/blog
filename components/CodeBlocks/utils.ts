import parseNumericRange from 'parse-numeric-range'
import type { Language } from 'prism-react-renderer'

const normalizeCode = (code: string = '') => code.replace(/\n+$/, '')

function isLanguage(language: string): language is Language {
  return language.length > 0
}

function normalizeLanguage(languageClass?: string): Language {
  if (languageClass != null) {
    const language = languageClass.replace('language-', '')

    if (isLanguage(language))
      return language
  }

  return 'typescript'
}

function normalizeLanguageName(language: string): string {
  switch (language) {
    case 'html':
      return 'HTML'
    case 'xml':
      return 'XML'
    case 'yml':
    case 'yaml':
      return 'YAML'
    case 'css':
      return 'CSS'
    case 'json':
      return 'JSON'
    case 'md':
    case 'markdown':
      return 'Markdown'
    case 'js':
    case 'javascript':
      return 'JavaScript'
    case 'ts':
    case 'typescript':
      return 'TypeScript'
    case 'coffee':
    case 'coffeescript':
      return 'CoffeeScript'
    case 'jsx':
    case 'tsx':
      return 'React'
    case 'objc':
    case 'objectivec':
      return 'Objective-C'
    default:
      return language.charAt(0).toUpperCase() + language.slice(1)
  }
}

function normalizeLines(expression: string): Set<number> {
  return new Set(parseNumericRange(expression))
}

export {
  normalizeCode,
  normalizeLanguage,
  normalizeLanguageName,
  normalizeLines,
}
