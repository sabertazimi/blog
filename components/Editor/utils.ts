function languageToFilepath(language?: string): string {
  switch (language) {
    case 'css':
      return '/styles.css'
    case 'js':
    case 'javascript':
      return '/App.js'
    case 'ts':
    case 'typescript':
      return '/App.ts'
    case 'jsx':
      return '/App.jsx'
    case 'tsx':
      return '/App.tsx'
    case 'vue':
      return '/src/App.vue'
    case undefined:
    default:
      return '/App.tsx'
  }
}

function normalizeFilepath(filename?: string, language?: string): string {
  if (filename == null || filename === '')
    return languageToFilepath(language)

  if (!filename.startsWith('/'))
    return `/${filename}`

  return filename
}

export { normalizeFilepath }
