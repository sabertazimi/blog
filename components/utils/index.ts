import classNames from 'classnames';

export { classNames };
