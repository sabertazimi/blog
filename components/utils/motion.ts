export { AnimatePresence, motion } from 'framer-motion';
export type { MotionProps } from 'framer-motion';
