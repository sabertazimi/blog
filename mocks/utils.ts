let navigatorClipboardMocked = false
const writeTextMock = jest.fn()

function mockNavigatorClipboard() {
  if (!navigatorClipboardMocked) {
    Object.defineProperty(globalThis.navigator, 'clipboard', {
      value: {
        writeText: writeTextMock,
      },
    })

    navigatorClipboardMocked = true
  }

  return { writeTextMock }
}

export { mockNavigatorClipboard }
