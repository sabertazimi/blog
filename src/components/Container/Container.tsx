import type { CSSProperties, HTMLProps, ReactNode } from 'react';
import React from 'react';

interface Props extends HTMLProps<HTMLDivElement> {
  className?: string;
  style?: CSSProperties;
  children: ReactNode;
}

const Container = ({
  className = '',
  style = {},
  children,
  ...props
}: Props): JSX.Element => (
  <div
    className={`container relative block h-full mx-auto my-0 p-auto ${className}`}
    style={{ ...style }}
    {...props}
  >
    {children}
  </div>
);

export default Container;
