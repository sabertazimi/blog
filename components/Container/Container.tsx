import { cx } from '@components/utils'
import type { CSSProperties, HTMLProps, ReactNode } from 'react'

interface Props extends HTMLProps<HTMLDivElement> {
  children: ReactNode
  className?: string
  style?: CSSProperties
}

const Container = ({ children, className, ...props }: Props): JSX.Element => (
  <div
    className={cx(
      'p-auto container relative mx-auto my-0 block h-full',
      className
    )}
    {...props}
  >
    {children}
  </div>
)

export default Container
