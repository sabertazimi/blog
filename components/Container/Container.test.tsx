import { axe } from 'jest-axe'
import Container from './Container'
import { render, screen } from '@/utils'

describe('Container', () => {
  it('should render correctly (snapshot)', () => {
    const { container } = render(
      <Container role="main">
        <h1>Container</h1>
      </Container>,
    )

    expect(container).toMatchSnapshot()
  })

  it('should render accessibility guidelines (AXE)', async () => {
    const { container } = render(
      <Container role="main">
        <h1>Container</h1>
      </Container>,
    )

    const a11y = await axe(container)

    expect(a11y).toHaveNoViolations()
  })

  it('should render children correctly', () => {
    render(
      <Container role="main">
        <h1>Container</h1>
      </Container>,
    )

    expect(screen.getByRole('main')).toContainElement(
      screen.getByText('Container'),
    )
  })
})
