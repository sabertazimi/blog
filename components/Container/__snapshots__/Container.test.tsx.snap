// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container should render correctly (snapshot) 1`] = `
<div>
  <div
    class="p-auto container relative mx-auto my-0 block h-full"
    role="main"
  >
    <h1>
      Container
    </h1>
  </div>
</div>
`;
